src/import_export_job/tasks.py

```diff
@@ -37,61 +37,45 @@ def get_format(job):
 
 
 def _run_import_job(import_job, dry_run=True):
-    change_job_status(import_job, "import", "1/5 Import started", dry_run)
+    def update_status(step, message):
+        change_job_status(import_job, "import", f"{step} {message}", dry_run)
+
+    def log_error(message, exception=None):
+        error_message = _(message)
+        if exception:
+            error_message += f": {exception}"
+        import_job.errors += error_message + "\n"
+        update_status("Error", error_message)
+        import_job.save()
+
+    update_status("1/4", "Import started")
     import_job.errors = ""
 
     model_config = ModelConfig(**importables[import_job.model])
-
     import_format = get_format(import_job)
 
-    # Copied from https://github.com/django-import-export/django-import-export/blob/3c082f98afe7996e79f936418fced3094f141c26/import_export/admin.py#L260 sorry  # noqa
     try:
-        data = import_job.file.read()
-
-        if isinstance(data, bytes):
-            data = data.decode("utf8")
+        data_chunks = []
+        with import_job.file.open('rb') as file:  # Open in binary mode
+            for chunk in file.chunks():
+                data_chunks.append(chunk.decode('utf-8'))
+        data = ''.join(data_chunks)
 
         dataset = import_format.create_dataset(data)
-
     except UnicodeDecodeError as e:
-        import_job.errors += _("Imported file has a wrong encoding: %s" % e) + "\n"
-        change_job_status(
-            import_job, "import", "Imported file has a wrong encoding", dry_run
-        )
-        import_job.save()
+        log_error("Imported file has a wrong encoding", e)
         return
-
     except Exception as e:
-        import_job.errors += _("Error reading file: %s") % e + "\n"
-        change_job_status(import_job, "import", "Error reading file", dry_run)
-        import_job.save()
+        log_error("Error reading file", e)
         return
```

Comment:
Catching the UnicodeDecodeError is very nice! However, I think catching the general exception afterwards is a bit too broad, especially because this exception could also be handled by the code that calls this function.

What I think would be nice is something like this:
```
except Exception as e:
  raise Exception("Error reading file: " + str(e)) from e
```

This will raise a new exception with all the context and information from the original exception. The name of the exception would be prepended with you own message.

I see that you are using the log_error function only for these exceptions. If you use the same pattern for the UnicodeDecodeError we could remove that function completely right?